### ISSUE TITLE
nvim, no mouse support in git-bash

### ISSUE THREAD

**Issue by fekir** [neovim/neovim#27062]
### Problem

`nvim` shoud have mouse support (`:set mouse=a`) by default.

This works as expected from `cmd.exe` and powershell (I'm testing on Windows 10, with the preinstalled powershell, not powershell core), but does not from `git-bash` (and cygwin, for what it's worth)

### Steps to reproduce

 * open `git-bash` (the shortcut executes `"C:\Program Files\Git\git-bash.exe" --cd-to-home`)
 * executi `'/c/Program Files/Neovim/bin/nvim.exe' --clean`
 * press `i` to enter insert mode
 * write something
 * click with the mouse to change the cursor positionor

NOTE, I've also tried unsetting `TERM, but it did not make a difference.

### Expected behavior

I would expect mouse support to behave the same in `git-bash`, powershell and `cmd.exe`

### Neovim version (nvim -v)

NVIM v0.9.5

### Vim (not Nvim) behaves the same?

yes (vim for windows, not vim of git-bash, and also after enabling mouse support)

### Operating system/version

Windows 10

### Terminal name/version

mintty 3.7.0 (x86_64-pc-msys)

### $TERM environment variable

xterm-256color

### Installation

Downloaded msi installer

**Comment by clason** [comment]
Your issue made it sound like you want a different default than we have.

If the issue is that the mouse doesn't work (regardless of option) in `git-bash` _only_, then the first step would be to verify that `git-bash` is even capable of receiving and forwarding mouse clicks.

**Comment by fekir** [comment]
> Your issue made it sound like you want a different default than we have.

No, my issue is that it does not work in `git-bash`.
What should I change in the description to make it more clear?

> [...] verify that git-bash is even capable of receiving and forwarding mouse clicks.

vim packaged for `git-bash` can handle mouse events, after setting `:set mouse=a`

**Comment by justinmk** [comment]
Thanks for the report. Can you try the development version? 

- The [Releases](https://github.com/neovim/neovim/releases/tag/nightly) page has pre-built archives for Linux/Windows/macOS. 
- Check `:version` to confirm that you are using the latest version.

**Comment by fekir** [comment]
@justinmk , tested with `NVIM v0.10.0-dev-2400+g8952a89db`, issue is still there, I'm not able to use the mouse

**Comment by ykhan21** [comment]
For the time being, you use a different terminal with bash.
Set "C:\Program Files\Git\bin\bash.exe -l -i" as the shell in alacritty or wezterm and you should be able to use bash+neovim on Windows with a mouse.

**Comment by ykhan21** [comment]
Also, I can't reproduce this on Windows 11 in Git Bash.
After setting `:se mouse+=a`, I am able to change the cursor position with the mouse.

Can you try if running `export MSYS=enable_pcon` before starting neovim allows it to work for you? Also, try if `winpty nvim` allows it to work.

**Comment by fekir** [comment]
Tested

 * Windows 10 version `22H2 Build 19045.4412` 
 * git bash (version `2.45.1.windows.1 (64-bit)` 
 * `nvim` version `v0.10.0`
 * with and without `export MSYS=enable_pcon`
 * with and without `winpty`
 * with `nvim --clean` and then enabling `:set mouse+=a`
 * `$TERM` is `xterm-256color`, I've also tried to unset it

but mouse is still not working

As additional info:
using `winpty vim` seems to disable mouse support in `vim` packaged for `git-bash`, while with `export MSYS=enable_pcon; vim` the  mouse still works in `vim` packaged for `git-bash`.


`winpty` also supports the parameter `--mouse`, but it does not seem to make any difference.


@ykhan21 Just to be clear, when you wrote `Git Bash`, do you mean that you have executed `git-bash.exe` from the installation folder of git for windows?

**Comment by ykhan21** [comment]
Yes, I started git-bash.exe by double clicking the exe in the installation folder.
After starting neovim, I run `:set mouse+=a`, which allows clicking to work in the mintty terminal.
I'm on Windows 11 with the latest git and neovim packages from scoop.
In my ~/.bashrc I have the line `export MSYS=winsymlinks:nativestrict`, which I believe includes `enable_pcon`. You tried `enable_pcon` but it didn't seem to work.

I think scoop's git is a portable thing when enables a few options in the git installer automatically. There could be some option that it's setting which you haven't.

**Comment by fekir** [comment]
I was able to test my setup on Windows 11 (VM), and mouse seems to be supported.

Same setup (same version of git-bash and nvim, both portable versions) on Windows 10 and it did not work.

I wonder what the difference could be...

**Comment by ykhan21** [comment]
When start git bash in Windows 11, is it using the mintty terminal or a Microsoft Terminal?

**Comment by fekir** [comment]
Mintty

**Comment by s3rb31** [comment]
> I was able to test my setup on Windows 11 (VM), and mouse seems to be supported.
> 
> Same setup (same version of git-bash and nvim, both portable versions) on Windows 10 and it did not work.
> 
> I wonder what the difference could be...

The difference probably lies somewhere in the implementation of conhost.exe. Replacing conhost.exe with OpenConsole.exe from Windows Terminal enables mouse support in neovim for me on Windows 10 22H2 inside mintty version 3.7.7.

https://github.com/microsoft/terminal/issues/1817
https://github.com/alacritty/alacritty/issues/1663
